koji-setup/mash.sh

```diff
@@ -16,30 +13,95 @@ MASH_TRACKER_DIR="$MASH_DIR"/latest
 MASH_DIR_OLD="$MASH_TRACKER_DIR".old
 MASH_DIR_NEW="$MASH_TRACKER_DIR".new
 
+create_dist_repos() {
+	local output_dir="${1}"
+
+	local work_dir="$(mktemp -d)"
+
+	local nvr_pkg_list="${work_dir}/nvr-pkg-list"
+	local bin_rpm_paths="${work_dir}/bin-rpm-paths"
+	local debuginfo_rpm_paths="${work_dir}/debuginfo-rpm-paths"
+	local src_rpm_paths="${work_dir}/src-rpm-paths"
+	local comps_file="${work_dir}/comps.xml"
+
+	sed -r -e 's|[^/]+/||' -e "s|^|${KOJI_DIR}/|" "${KOJI_REPO_PATH}/${BUILD_ARCH}/pkglist" > "${bin_rpm_paths}"
+	sed -r -e 's|^([^/]+/){2}||' -e 's|/[^/]+/[^/]+$||' "${KOJI_REPO_PATH}/${BUILD_ARCH}/pkglist" | sort -u > "${nvr_pkg_list}"
```

Comment:
Is the `sort -u` needed here?

Also, as a simplification, I would prefer using `cut` instead:

```
cut -d/ -f3-5 "${KOJI_REPO_PATH}/${BUILD_ARCH}/pkglist" > "${nvr_pkg_list}"
```

Reply:
The `sort` so much, no, but the `-u`, yes, since otherwise it would repeat NVRs.

Ah, nice trick with the cut.  Changed.